src/ng-select/ng-select.component.ts

```diff
@@ -269,7 +274,7 @@ export class NgSelectComponent implements OnInit, OnDestroy, OnChanges, AfterVie
     }
 
     open() {
-        if (this.isDisabled || this.isOpen) {
+        if (this.isDisabled || this.isOpen || this.isMaximumNumberOfSelectedItemsReaches()) {
```

Comment:
You can remove that method and use one prod itemsList directly 
```js
if (this.isDisabled || this.isOpen || this.itemsList.isMaximumNumberOfSelectedItemsReaches()) {
```

Reply:
I think in this case it should be `this.itemsList.isMaximumNumberOfSelectedItemsReached()` then